jekyll/_cci2/language-dart.md

```diff
@@ -0,0 +1,435 @@
+
+## Config Walkthrough
+
+The first section of the file defines common items. The order in which things are defined in a CircleCI configuration doesn't matter - we have laid it out this way for readability.
+
+```yaml
+version: 2.1
+
+#####################
+# Common Definitions
+#####################
+
+# Orb declarations
+orbs:
+  win: circleci/windows@2.4.0
+```
+
+- **Config Version**: Current is 2.1. Unless using legacy server install or legacy jobs and workflows, we highly suggest using 2.1.
+- **Orbs**: CircleCI offers [orbs](https://circleci.com/orbs/registry/), which are packaged, templatized, and reusable configuration. Here, we include the [Windows orb](https://circleci.com/orbs/registry/orb/circleci/windows) to make use of a pre-defined execution environment later on.
+
+{% raw %}
+
+```yaml
+commands:
+  dependencies:
+    description: "Download dependencies and setup global packages"
+    parameters:
+      shell:
+        type: string
+        default: "/bin/bash --login -eo pipefail"
+      pub-cache:
+        type: string
+        default: "~/.pub-cache"
+    steps:
+      - checkout
+      - restore_cache:
+          keys:
+            - v1.4-dependencies-{{ arch }}-{{ checksum "pubspec.lock" }}
+            - v1.4-dependencies-{{ arch }}-
+      - run:
+          name: Download deps
+          shell: << parameters.shell >>
+          command: pub get
+      - run:
+          name: Get junitreporter
+          shell: << parameters.shell >>
+          command: pub global activate junitreport
+      - save_cache:
+          key: v1.4-dependencies-{{ arch }}-{{ checksum "pubspec.lock" }}
+          paths:
+            - .dart_tool
+            - << parameters.pub-cache >>
+
+  native-build:
+    description: "Runs the dart2native command to build native executable for machine. Artifacts executable"
+    parameters:
+      shell:
+        type: string
+        default: "/bin/bash --login -eo pipefail"
+    steps:
+      - run:
+          name: Native compile
+          shell: << parameters.shell >>
+          command: dart2native bin/server.dart -o circleci_dart_demo.exe
+      - store_artifacts:
+          path: circleci_dart_demo.exe
+```
+
+{% endraw %}
+
+- CircleCI offers [reusable configuration syntax]({{ site.baseurl }}/2.0/reusing-config), which is what our orbs are built off of. Under `commands`, we define a few parameterized commands that run the same sequence of steps.
+    - `dependencies` checks out the code and restores/saves a dependency cache (Dart packages downloaded from last build).
+    - `native-build` uses the `dart2native` command to compile a native executable for each operating system and then artifacts it, making it available to download from the job details.
+    - In both commands, we take parameters to override a cache path and shell for different jobs. Dart's package cache is in a different location in Windows, and it also uses a different shell (Powershell).
+
+```yaml
+######################
```

Comment:
I think I would recommend removing the large comment blocks - the sections are already broken up, and none of our language guides use them.